ui/src/tempVars/components/TemplateControlDropdown.tsx

```diff
@@ -0,0 +1,119 @@
+import React, {PureComponent} from 'react'
+
+import Dropdown from 'src/shared/components/Dropdown'
+import SimpleOverlay from 'src/shared/components/SimpleOverlay'
+import EditTemplateVariable from 'src/tempVars/components/EditTemplateVariable'
+import {calculateDropdownWidth} from 'src/dashboards/constants/templateControlBar'
+import Authorized, {isUserAuthorized, EDITOR_ROLE} from 'src/auth/Authorized'
+import {Template, Source} from 'src/types'
+
+interface Props {
+  template: Template
+  meRole: string
+  isUsingAuth: boolean
+  source: Source
+  onSelectTemplate: (id: string) => void
+  onCreateTemplate: (template: Template) => Promise<any>
```

Comment:
1. why are these `any`? could you make it more specific with an `|`?
1. same with the two below